core/domain/state_domain.py

```diff
@@ -916,8 +940,1085 @@ def get_all_outcomes(self) -> List[Outcome]:
+                        )
+                    except Exception:
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsWithinTolerance\' contains string '
+                            f'value - \'{rule_value_x}\', \'{rule_value_tol}\''
+                        )
+
+                if rule_spec.rule_type == 'IsGreaterThan':
+                    try:
+                        rule_value = float(rule_spec.inputs['x'])
+                        self._set_lower_and_upper_bounds(
+                            range_var, rule_value,
+                            upper_infinity, False, False
+                        )
+                    except Exception:
+                        rule_value = rule_spec.inputs['x']
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsGreaterThan\' contains string '
+                            f'value - \'{rule_value}\''
+                        )
+
+                if rule_spec.rule_type == 'IsInclusivelyBetween':
+                    rule_value_a = rule_spec.inputs['a']
+                    rule_value_b = rule_spec.inputs['b']
+                    try:
+                        rule_value_a = float(rule_value_a)
+                        rule_value_b = float(rule_value_b)
+                        # `a` should not be greater than `b` in
+                        # `IsInclusivelyBetween` rule.
+                        if rule_value_a > rule_value_b:
+                            raise utils.ValidationError(
+                                f'The rule \'{rule_spec_index}\' of answer '
+                                f'group \'{ans_group_index}\' having '
+                                f'rule type \'IsInclusivelyBetween\' '
+                                f'have `a` value greater than `b` value '
+                                f'in NumericInput interaction.'
+                            )
+                        self._set_lower_and_upper_bounds(
+                            range_var, rule_value_a,
+                            rule_value_b, True, True
+                        )
+                    except Exception:
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsInclusivelyBetween\' contains string '
+                            f'value - \'{rule_value_a}\', \'{rule_value_b}\''
+                        )
+
+                for range_ele in ranges:
+                    if self._is_enclosed_by(range_var, range_ele):
```

Comment:
shouldn't this be wrapped in a `should_check_range_enclosure` function? We don't want this hardcoded to check range enclosure for every rule type by default. If we want it to run for every rule type that we currently have in the codebase, then we should hardcode a list of rule types.

Reply:
Yep! Updated